Src/FluentAssertions/Primitives/NullableEnumAssertions.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a nullable <typeparamref name="TEnum"/> is in the expected state.
+    /// </summary>
+    public class NullableEnumAssertions<TEnum> : NullableEnumAssertions<TEnum, NullableEnumAssertions<TEnum>>
+        where TEnum : struct, Enum
+    {
+        public NullableEnumAssertions(TEnum? subject)
+            : base(subject)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Contains a number of methods to assert that a nullable <typeparamref name="TEnum"/> is in the expected state.
+    /// </summary>
+    public class NullableEnumAssertions<TEnum, TAssertions> : EnumAssertions<TEnum, TAssertions>
+        where TEnum : struct, Enum
+        where TAssertions : NullableEnumAssertions<TEnum, TAssertions>
+    {
+        public NullableEnumAssertions(TEnum? subject)
+            : base(subject)
+        {
+        }
+
+        public new TEnum? Subject => SubjectInternal;
+
+        /// <summary>
+        /// Asserts that a nullable <typeparamref name="TEnum"/> value is not <c>null</c>.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])"/> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndWhichConstraint<TAssertions, TEnum> HaveValue(string because = "", params object[] becauseArgs)
```

Comment:
🤔 I noticed that we have `HaveValue` and `NotBeNull`. I guess this is just for convenience as well as consisteny with other similar cases, right?

Reply:
Yes, I included both for consistency with other `NullableSomethingAssertions`.